sdk/spring/spring-cloud-azure-actuator/src/main/java/com/azure/spring/cloud/actuator/cosmos/CosmosHealthIndicator.java

```diff
@@ -45,18 +47,28 @@ public CosmosHealthIndicator(CosmosAsyncClient cosmosAsyncClient, String databas
 
     @Override
     protected void doHealthCheck(Builder builder) {
-        CosmosDatabaseResponse response = this.cosmosAsyncClient.getDatabase(database)
-            .read()
-            .block(timeout);
-
-        if (response != null) {
-            LOGGER.info("The health indicator cost {} RUs, cosmos uri: {}, dbName: {}",
-                response.getRequestCharge(), endpoint, database);
-        }
-        if (response == null) {
-            builder.down();
+        if (database == null) {
+            builder.status(Status.UNKNOWN).withDetail("Database not configured",
+                "The option of `spring.cloud.azure.cosmos.database` is not configured!");
         } else {
-            builder.up().withDetail("database", response.getProperties().getId());
+            try {
+                CosmosDatabaseResponse response = this.cosmosAsyncClient.getDatabase(database)
+                                                                        .read()
+                                                                        .block(timeout);
+
+                if (response != null) {
+                    LOGGER.info("The health indicator cost {} RUs, cosmos uri: {}, dbName: {}",
+                        response.getRequestCharge(), endpoint, database);
+                    builder.withDetail("RUs", response.getRequestCharge());
+                    builder.withDetail("CosmosUri", endpoint);
+                    builder.up().withDetail("database", database);
```

Comment:
It's better to put `builder.up()` at the beginning.

Reply:
How about doing like this: 
```java
  builder.up()
      .withDetail("", "")
      .withDetail("", "");
```
Same to all other places.

Reply:
yes, this looks better.